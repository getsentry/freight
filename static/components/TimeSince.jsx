import PropTypes from 'prop-types';
import React from 'react';
import {formatDistanceToNowStrict} from 'date-fns';

class TimeSince extends React.Component {
  static propTypes = {
    date: PropTypes.any.isRequired,
  };

  componentDidMount() {
    const delay = 2600;

    this.ticker = setInterval(this.ensureValidity, delay);
  }

  componentWillUnmount() {
    if (this.ticker) {
      clearInterval(this.ticker);
      this.ticker = null;
    }
  }

  ensureValidity = () => {
    // TODO(dcramer): this should ensure we actually *need* to update the value
    this.forceUpdate();
  };

  render() {
    const date = new Date(this.props.date);

    return (
      <time dateTime={date.toISOString()}>
        {formatDistanceToNowStrict(date, {addSuffix: true})}
      </time>
    );
  }
}

export default TimeSince;
