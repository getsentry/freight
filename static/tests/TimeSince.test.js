import React from 'react';
import {shallow} from 'enzyme';
import TimeSince from '../components/TimeSince.jsx';

test('TimeSince Snapshot', () => {
  const date = 1482363367071;
  const wrapper = shallow(<TimeSince date={date} />);
  expect(wrapper).toMatchSnapshot();
});
