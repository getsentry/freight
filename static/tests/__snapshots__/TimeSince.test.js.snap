// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSince Snapshot 1`] = `
<time
  dateTime="2016-12-21T23:36:07.071Z"
>
  3 years ago
</time>
`;
